src/engine/cluster/pilot_quantum_executor.py

```diff
@@ -25,9 +16,24 @@ def __init__(self, cluster_config=None):
         super().__init__()
         self.cluster_config = cluster_config or {}
         self.type = self.cluster_config["config"]["type"]
-        self.pilot, self.client = initialize_client(self.cluster_config["config"])
+        self.pilot, self.client = self.initialize_client(self.cluster_config["config"])
+
+    def initialize_client(self, cluster_config):
+        pilot_compute_description = cluster_config
+        execution_engine = ExecutionEngine(cluster_config["type"])
+        working_directory = cluster_config["working_directory"]
+    
+        pcs = PilotComputeService(execution_engine=execution_engine, working_directory=working_directory)
+        pilot = pcs.create_pilot(pilot_compute_description=pilot_compute_description)
+        pilot.wait()
+        dask_ray_client = pilot.get_client()
+        return pilot, dask_ray_client
 
     def close(self):
+        if self.type == "dask":
+            self.client.close()
+        elif self.type == "ray":
+            ray.shutdown()
```

Comment:
the pilot.cancel which abstracts dask/ray engine should clean it. this seems like leaking pilot-abstraction into mini-apps.

Reply:
The pilot-cancel already does this - https://github.com/radical-cybertools/pilot-quantum/blob/ray-v2/pilot/plugins/ray_v2/cluster.py#L136